tests/test_logs.py

```diff
@@ -0,0 +1,197 @@
+    """Ensure that messages above maximum are not logged"""
+    log_filter = smartsim.log.LowPassFilter("WARN")
+
+    faux_out_stream = io.StringIO()
+
+    logger = logging.getLogger("test_level_filter_warn")
+    logger.addHandler(logging.StreamHandler(faux_out_stream))
+    logger.addFilter(log_filter)
+
+    logger.debug("debug")
+    logger.info("info")
+    logger.warning("warn")
+    logger.error("error")
+    logger.exception("exception")
+
+    logged_messages = faux_out_stream.getvalue().split("\n")
+    assert "debug" in logged_messages
+    assert "info" in logged_messages
+    assert "warn" in logged_messages
+    assert "error" not in logged_messages
+    assert "exception" not in logged_messages
+
+
+def test_add_exp_loggers(test_dir):
+    """Ensure that expected loggers are added"""
+    # test_dir = fileutils.make_test_dir()
+    faux_out_stream = io.StringIO()
+
+    logger = logging.getLogger("smartsim_test_add_exp_loggers")
+    logger.addHandler(logging.StreamHandler(faux_out_stream))
+
+    out_file = pathlib.Path(test_dir) / "smartsim.out"
+    err_file = pathlib.Path(test_dir) / "smartsim.err"
+
+    filter_fn = lambda x: True
+
+    smartsim.log.log_to_file(str(out_file), logger=logger, log_filter=filter_fn)
+    smartsim.log.log_to_file(str(err_file), "WARN", logger)
+
+    logger.debug("debug")
+    logger.exception("exception")
+
+    assert out_file.exists()
+    assert out_file.is_file()
+
+    assert err_file.exists()
+    assert err_file.is_file()
+
+
+def test_get_logger(test_dir: str):
+    """Ensure the correct logger type is instantiated"""
+    logger = smartsim.log.get_logger("SmartSimTest", "INFO")
+    assert isinstance(logger, smartsim.log.ContextAwareLogger)
+
+
+def test_exp_logs(test_dir: str):
+    """Ensure that experiment loggers are added when context info exists"""
+    test_dir = pathlib.Path(test_dir)
+    test_dir.mkdir(parents=True, exist_ok=True)
+
+    token = smartsim.log.ctx_exp_path.set(test_dir)
+
+    logger = smartsim.log.get_logger("SmartSimTest", "INFO")
+
+    faux_out_stream = io.StringIO()
+    logger.addHandler(logging.StreamHandler(faux_out_stream))
+
+    log_msg = "testing in a test!"
+    err_msg = "erroring in a test!"
+    logger.info(log_msg)
+    logger.error(err_msg)
+
+    # ensure that the default stream is written to
+    logged = faux_out_stream.getvalue()
+    assert log_msg in logged
+    assert err_msg in logged
+
+    out_file, err_file = smartsim.log.get_exp_log_paths()
+
+    out_content = out_file.read_text()
+    err_content = err_file.read_text()
+
+    # ensure the low-pass filter logs non-errors to out file
+    assert log_msg in out_content
+    assert err_msg not in out_content
+    assert str(test_dir) in out_content
+
+    # ensure the errors are logged to err file
+    assert err_msg in err_content
+    assert log_msg not in err_content
+    assert str(err_msg) in err_content
+
+    smartsim.log.ctx_exp_path.reset(token)
```

Comment:
Quick sanity check here: the `smartsim.log.ctx_exp_path` state will not leak outside of this test if any of the previous `assert smts` fail right?

Reply:
same sanity check for `test_exp_logs_without_ctx`

Reply:
wrapped in try/finally to be sure. great catch.